pyk/src/pyk/subst.py

```diff
@@ -46,6 +55,51 @@ def replace(term):
 
         return bottom_up(replace, term)
 
+    def unapply(self, term: KInner) -> KInner:
+        new_term = term
+        for var_name in self:
+            new_term = replaceAnywhereWith((self[var_name], KVariable(var_name)), new_term)
+        return new_term
+
+
+def extract_subst(term: KInner) -> Tuple[Subst, KInner]:
+
+    def _subst_for_terms(term1: KInner, term2: KInner) -> Optional[Subst]:
+        if type(term1) is KVariable and not type(term2) is KVariable:
+            return Subst({term1.name: term2})
+        if type(term2) is KVariable and not type(term1) is KVariable:
+            return Subst({term2.name: term1})
```

Comment:
And here we are filtering for only substitutions that are not of the form `KVariable == KVariable`.

So it could be that the extra check for whether one side is a token (mentioned here https://github.com/runtimeverification/k/pull/2586/files#r869556888) actually happens here, so that when you have `KVariable == KToken`, it doesn't get counted as a substitution at all by `extract_subst`.